### ISSUE TITLE
Details on How to Create and Run Local Migrations

### ISSUE THREAD

**Issue by bsbechtel** [nrwl/nx#10617]
### Documentation issue

- [ ] Reporting a typo
- [ ] Reporting a documentation bug
- [X] Documentation improvement
- [X] Documentation feedback

### Is there a specific documentation page you are reporting?

Enter the URL or documentation section here.
https://nx.dev/packages/nx-plugin/generators/migration

### Additional context or description
I am trying to develop a migration for a library, which I would like to apply to multiple apps within the same monorepo. I am able to successfully generate a migration with the command `npx nx g @nrwl/nx-plugin:migration "<name your migration here>" --project=<library-name>`, and add the migration to the monorepo's `migrations.json` file, but I can't get it to successfully run.

It appears that Nx is executing the entire npm install and node resolution process in a temporary directory, which means that any local libraries are excluded from the migration process. I know that there must be a way to run migrations locally to facilitate the development process. It would be nice to have this documented. At a minimum, I'm looking for information on how to get myself unblocked.

Provide any additional details here as needed.

**Comment by rishabhmhjn** [comment]
Following this as I am also looking at creating migrations frequently to be used internal in our monoreps for use-cases such as moving/updating imports etc

**Comment by isaacplmann** [comment]
If you want to run a migration only on the repo where it is created, you want a generator, not a migration.  A migration is just a generator with no inputs.

The reason the migration test script creates a new temp folder to run tests against is because migrations are intended to be executed on code in someone else's repo that you don't control.
